src/stream.rs

```diff
@@ -119,3 +135,51 @@ pub(crate) async fn tripwire_handler(closed: bool) {
     })
     .await
 }
+
+/// Calls function F after an item from stream has been processed
+/// by the consumer.
+///
+/// This assumes that the stream is being consumed in a loop, so when
+/// the consumer get's an item and then comes for another one
+/// we assume that it has processed the first item.
+/// This should hold in most cases. One example of when this doesn't
+/// hold is when the consumer batches items, or if it goes away and
+/// does something unrelated to the item which it has already processed.
+/// In such cases this construct can still be used, but it won't be accurate.
```

Comment:
What if the stream idles for a prolonged time? I think the assumption does not hold then, and I deem this would be a critical case - one we want to properly support.

Reply:
I think this is also related to my thoughts that we should just have metrics for in/out rather than trying to determine when an event has been "processed" by a transform.

Reply:
> What if the stream idles for a prolonged time? 

It holds since the the stream will be polled in either case, so the emit will happen.

> I think this is also related to my thoughts that we should just have metrics for in/out rather than trying to determine when an event has been "processed" by a transform. 

I do agree, but I also don't think that we should wait for in/out to be added.

This is not perfect, but it works in our cases so I'm for going forward with this.